File: double_dispatch/java/01_polymorphic/GameRunner.java
public class GameRunner {
	public static void main(String[] args) {
		Beatable paper = new Paper();
		Beatable rock = new Rock();
		Beatable scissor = new Scissor();
		
		System.out.println("paper.beats(paper)? " + paper.beats(paper));
		System.out.println("paper.beats(scissor)? " + paper.beats(scissor));
    System.out.println("paper.beats(rock)? " + paper.beats(rock));
		System.out.println("scissor.beats(paper)? " + scissor.beats(paper));
    System.out.println("scissor.beats(scissor)? " + scissor.beats(scissor));
    System.out.println("scissor.beats(rock)? " + scissor.beats(rock));
    System.out.println("rock.beats(paper)? " + rock.beats(paper));
    System.out.println("rock.beats(scissor)? " + rock.beats(scissor));
    System.out.println("rock.beats(rock)? " + rock.beats(rock));
	}
}

File: 02_double_dispatch/c#/02_after_pattern_matching/GameRunner.cs
using System;

public class GameRunner {
  public static void Main(string[] args) {
    var paper = new Paper();
    var rock = new Rock();
    var scissor = new Scissor();

    Console.WriteLine("paper.Beats(paper) = " + paper.Beats(paper));
    Console.WriteLine("paper.Beats(scissor) = " + paper.Beats(scissor));
    Console.WriteLine("paper.Beats(rock) = " + paper.Beats(rock));
    Console.WriteLine("scissor.Beats(paper) = " + scissor.Beats(paper));
    Console.WriteLine("scissor.Beats(scissor) = " + scissor.Beats(scissor));
    Console.WriteLine("scissor.Beats(rock) = " + scissor.Beats(rock));
    Console.WriteLine("rock.Beats(paper) = " + rock.Beats(paper));
    Console.WriteLine("rock.Beats(scissor) = " + rock.Beats(scissor));
    Console.WriteLine("rock.Beats(rock) = " + rock.Beats(rock));
  }
}
